File: C/FirstOccurance.c
#include <stdio.h>

int firstOccurrence(int arr[], int n, int key) {
    int low = 0, high = n - 1, result = -1;
    while (low <= high) {
        int mid = low + (high - low) / 2;
        
        if (arr[mid] == key) {
            result = mid;       
            high = mid - 1;  
        }
        else if (arr[mid] > key) {
            high = mid - 1;
        }
        else {
            low = mid + 1;
        }
    }
    return result;
}

int main() {
    int arr[] = {1, 2, 2, 2, 3, 4};
    int n = sizeof(arr) / sizeof(arr[0]);
    int key = 2;
    
    int index = firstOccurrence(arr, n, key);
    if (index != -1)
        printf("First occurrence of %d is at index %d\n", key, index);
    else
        printf("%d not found in array\n", key);
    
    return 0;
}

File: Python/FirstOccurance.py
def first_occurrence(arr, key):
    low, high = 0, len(arr) - 1
    result = -1
    
    while low <= high:
        mid = (low + high) // 2
        
        if arr[mid] == key:
            result = mid        
            high = mid - 1     
        elif arr[mid] > key:
            high = mid - 1
        else:
            low = mid + 1
    
    return result


# Example usage
arr = [1, 2, 2, 2, 3, 4]
key = 2
index = first_occurrence(arr, key)

if index != -1:
    print(f"First occurrence of {key} is at index {index}")
else:
    print(f"{key} not found in array")
